crates/goose/src/providers/google.rs

```diff
@@ -61,16 +62,21 @@ impl GoogleProvider {
     }
 
     async fn post(&self, payload: Value) -> Result<Value, ProviderError> {
-        let url = format!(
-            "{}/v1beta/models/{}:generateContent?key={}",
-            self.host.trim_end_matches('/'),
-            self.model.model_name,
-            self.api_key
-        );
+        let base_url = Url::parse(&self.host)
+            .map_err(|e| ProviderError::RequestFailed(format!("Invalid base URL: {e}")))?;
+
+        let url = base_url
+            .join(&format!(
```

Comment:
nit: above its 
```rust
let path =...

base_url.join(&path)
```
vs where we do the `format!` inline to join lets stick with one consistent approach? 
```rust
base_url.join(format!(...))
```